ludwig/models/trainer.py

```diff
@@ -1418,22 +1418,22 @@ def check_progress_on_validation(
             should_break = True
         return should_break
 
-    def set_epochs_to_1_or_quit(self, signum, frame):
+    def set_steps_to_1_or_quit(self, signum, frame):
+        """Custom SIGINT handler used to elegantly exit training.
+
+        A single SIGINT will stop training after the next training step. A second SIGINT will stop training immediately.
+        """
         if not self.received_sigint:
-            self.epochs = 1
+            self.total_steps = 1
             self.received_sigint = True
-            logger.critical("\nReceived SIGINT, will finish this epoch and then conclude " "the training")
-            logger.critical("Send another SIGINT to immediately interrupt the process")
+            logger.critical("\nReceived SIGINT, will finish this training step and then conclude training.")
+            logger.critical("Send another SIGINT to immediately interrupt the process.")
         else:
             logger.critical("\nReceived a second SIGINT, will now quit")
             if self.original_sigint_handler:
                 signal.signal(signal.SIGINT, self.original_sigint_handler)
             sys.exit(1)
 
-    def quit_training(self, signum, frame):
-        logger.critical("Received SIGQUIT, will kill training")
-        sys.exit(1)
-
```

Comment:
Just curious, why removing this? Was it dangling / not used?

Reply:
But it was there for a reason, if I remember correctly that function was associated with another signal instead of sigint to still give users a mechanism to kill with one single signal. Probably it got lost during the porting. I would add a sigquit handler like we do for sigint. Not a deal breaker though